src/git.rs

```diff
@@ -11,8 +12,48 @@ pub fn git_fetch(root: &Path) -> Result<(), git2::Error> {
     Ok(())
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct BranchInfo {
+    name: Option<String>,
+    commit_hash: String,
+    time: i64,
+    message: Option<Vec<String>>,
+}
+
+fn get_branch_info(
+    branch_struct: Result<(Branch, BranchType), git2::Error>,
+) -> Result<BranchInfo, git2::Error> {
+    let branch = branch_struct?.0;
+    let name = branch.name()?.map(String::from);
+
+    let branch_commit = branch.into_reference().peel_to_commit()?;
+    let message = branch_commit.message().map(|s| {
+        s.split_terminator("\n")
+            .map(String::from)
+            .collect::<Vec<String>>()
+    });
+
+    Ok(BranchInfo {
+        name,
+        commit_hash: branch_commit.id().to_string(),
+        time: branch_commit.time().seconds(),
+        message,
+    })
+}
+
+pub fn git_list_branches(root: &Path) -> Result<Vec<BranchInfo>, git2::Error> {
+    let repo = Repository::open(root)?;
+    let git_branches: Result<Vec<BranchInfo>, git2::Error> = repo
+        .branches(Some(BranchType::Local))?
```

Comment:
As discussed separately, I think we actually want to list branches of the origin remote, not the local ones.

The local ones (if they even exist) will not be updated by the `fetch` API. Only the remote ones are (and these are actually the local view of the remote branches ...)

Reply:
yep done